spec/lib/granite/action/precondition_spec.rb

```diff
@@ -4,7 +4,7 @@
       description 'Test description'
 
       def call(expected_title:, **)
-        expected_title == title
+        respond_to_missing?(:title) && expected_title == title
```

Comment:
Improving spec coverage

Reply:
@andriusch `#respond_to_missing?` shouldn't be used directly.

Quoting Ruby docs:
> DO NOT USE THIS DIRECTLY.

Source: https://www.rubydoc.info/stdlib/core/Object:respond_to_missing%3F

Reply:
That's the only way to test `def respond_to_missing?` though, since if I use `respond_to?` it actually delegates to `@context` since `BasicObject` does not define `respond_to?` method.
Which obviously raises a question whether we actually need `def respond_to_missing?`?

Reply:
https://thoughtbot.com/blog/always-define-respond-to-missing-when-overriding

Reply:
That's only if you're not using `BasicObject` as a parent class, in our case:

```ruby
class ProxyObject < BasicObject
  def initialize(object)
    @object = object
  end

  def method_missing(method, ...)
     @object.__send__(method, ...)
  end
end

proxy = ProxyObject.new("hello world")
proxy.respond_to?(:split) #=> true
proxy.split(' ') #=> ["hello", "world"]
```

Compare to when using `Object` as a parent class:

```ruby
class ProxyObject
  def initialize(object)
    @object = object
  end

  def method_missing(method, ...)
     @object.__send__(method, ...)
  end
end

proxy = ProxyObject.new("hello world")
proxy.respond_to?(:split) #=> false
proxy.split(' ') #=> ["hello", "world"]
```

This is because `BasicObject` does not define `respond_to?`, so `respond_to?` is actually proxied to `@object`.
Now that I think about it - it creates another problem:

```ruby
Granite::Action::Precondition.new(Object.new).respond_to?(:call) #=> false
```